spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mail/MailSenderAutoConfiguration.java

```diff
@@ -51,7 +50,7 @@
 	MailProperties properties;
 
 	@Bean
-	public JavaMailSender mailSender() {
+	public JavaMailSenderImpl mailSender() {
```

Comment:
I get why you want that but I'd like to commit that separately. I'll push that to master in a minute, please rebase.